### PR TITLE
Introduce player action state and shot state (to be implemented), add…

### PR DESCRIPTION
**Pull request by patkkka**
… Players class to handle two players

Please focus on PlayerActionState and Players class.

### CODE REVIEW

File: src/main/java/com/spanish_inquisition/battleship/server/game_states/PlacingShipsState.java [hunk 1]
```diff
@@ -10,29 +13,25 @@
...
                 }
             }
         }
-        return new EndState();
+        return new PlayerActionState(players, requestBus);
     }
 
     private void resolvePlayerFleetValidation(Player player, List<Player> notReadyPlayers) {
-        if(setFleetIfIsValid(player)) {
+        if (setFleetIfIsValid(player)) {
```
Comment (agawda) [inline src/main/java/com/spanish_inquisition/battleship/server/game_states/PlacingShipsState.java]: I would split it and actually send an already valid fleet here just to place it.

File: src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+public class ShotState extends GameState {
+    public ShotState(Players players, MessageBus requestBus) {
+        super(players, requestBus);
+    }
+
+    @Override
+    public GameState transform() {
+        if (requestBus.haveMessageFromSender(players.getCurrentPlayer().getPlayerId())) {
+            //TODO get message &
```
Comment (agawda) [inline src/main/java/com/spanish_inquisition/battleship/server/game_states/ShotState.java]: &? :smirk_cat:

File: src/test/java/com/spanish_inquisition/battleship/server/EndStateTest.java [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
 
+import com.spanish_inquisition.battleship.server.game_states.EndState;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 public class EndStateTest {
     @Test
     public void checkIfGameIsNotRunning() {
-        Assert.assertFalse(new EndState().isGameRunning());
+        Assert.assertFalse(new EndState(null, null).isGameRunning());
```
Comment (agawda) [inline src/test/java/com/spanish_inquisition/battleship/server/EndStateTest.java]: :crying_cat_face:

File: src/test/java/com/spanish_inquisition/battleship/server/PlacingShipsStateTest.java [hunk 4]
```diff
@@ -34,7 +36,7 @@ public void initialize() {
         }
         new Thread(() -> assignSocketAndNameTo("Name 1")).run();
         new Thread(() -> assignSocketAndNameTo("Name 2")).run();
-        players = new ArrayList<>();
+        players = new Players();
         for(int i = 1; i < 3; i++) {
```
Comment (agawda) [inline src/test/java/com/spanish_inquisition/battleship/server/PlacingShipsStateTest.java]: Magic numbers.

File: src/main/java/com/spanish_inquisition/battleship/server/Players.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
+package com.spanish_inquisition.battleship.server;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class Players {
```
Comment (arturAR) [inline src/main/java/com/spanish_inquisition/battleship/server/Players.java]: Maybe Enum?

File: src/test/java/com/spanish_inquisition/battleship/server/EndStateTest.java (hunk 3 above)
Comment (arturAR) [inline src/test/java/com/spanish_inquisition/battleship/server/EndStateTest.java]: Mocks?

File: src/test/java/com/spanish_inquisition/battleship/server/GameStateTest.java [hunk 6]
```diff
@@ -1,12 +1,13 @@
...
 
+import com.spanish_inquisition.battleship.server.game_states.GameState;
 import org.testng.Assert;
 import org.testng.annotations.Test;
 
 public class GameStateTest {
     @Test
     public void checkIfGameIsRunning() {
-        Assert.assertTrue(new GameState() {
+        Assert.assertTrue(new GameState(null, null) {
```
Comment (arturAR) [inline src/test/java/com/spanish_inquisition/battleship/server/GameStateTest.java]: Mocks?

File: src/main/java/com/spanish_inquisition/battleship/server/game_states/PlayerActionState.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+public class PlayerActionState extends GameState {
+
+    public PlayerActionState(Players players, MessageBus requestBus) {
+        super(players, requestBus);
+    }
+
+    @Override
+    public GameState transform() {
+        requestBus.addMessage(SERVER_ID, players.getCurrentPlayer().getPlayerId(), Header.PLAYER_TURN.name());
```
Comment (piotr921) [inline src/main/java/com/spanish_inquisition/battleship/server/game_states/PlayerActionState.java]: 👍

**Pull request merged**